earthaccess/auth.py

```diff
@@ -316,7 +316,7 @@ def _get_credentials(
         if user_token is not None:
             self.token = {"access_token": user_token}
             self.authenticated = True
-        if username is not None and password is not None:
+        elif username is not None and password is not None:
```

Comment:
This is a great catch, I don't like these `if` but we have n different ways of signing in.